File: app/src/main/java/com/example/findmeinlol/view/MainActivity.java
package com.example.findmeinlol.view;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProvider;

import android.content.Intent;
import android.os.Bundle;
import android.text.InputType;

import com.example.findmeinlol.R;
import com.example.findmeinlol.databinding.ActivityMainBinding;
import com.example.findmeinlol.viewmodel.MainViewModel;

public class MainActivity extends AppCompatActivity {

    private MainViewModel mainViewModel;
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        mainViewModel = new ViewModelProvider(this,
                (ViewModelProvider.Factory) ViewModelProvider.AndroidViewModelFactory
                        .getInstance(getApplication())).get(MainViewModel.class);

        binding.setLifecycleOwner(this);
        binding.setMainViewModel(mainViewModel);
        binding.mainSearchView.clearFocus();
        binding.mainSearchView.setInputType(InputType.TYPE_NULL);

        setEventListener();
    }

    private void setEventListener() {
        binding.mainSearchView.setOnClickListener(v -> {
            binding.mainSearchView.clearFocus();
            Intent intent = new Intent(MainActivity.this, SearchActivity.class);
            startActivity(intent);
        });

        binding.mainSearchView.setOnQueryTextFocusChangeListener((v, hasFocus) -> {
            if (hasFocus) {
                binding.mainSearchView.clearFocus();
                Intent intent = new Intent(MainActivity.this, SearchActivity.class);
                startActivity(intent);
            }
        });

        binding.mainTextView.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, LckMatchActivity.class);
            startActivity(intent);
        });
    }
}

File: app/src/main/java/com/example/findmeinlol/view/MainActivity.kt
package com.example.findmeinlol.view

import android.content.Intent
import android.os.Bundle
import android.text.InputType
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import com.example.findmeinlol.R
import com.example.findmeinlol.databinding.ActivityMainBinding
import com.example.findmeinlol.viewmodel.MainViewModel

class MainActivity : AppCompatActivity() {
    private lateinit var mainViewModel: MainViewModel
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        mainViewModel = ViewModelProvider(
            this,
            ViewModelProvider.AndroidViewModelFactory.getInstance(application)
        )
            .get(MainViewModel::class.java)

        binding.lifecycleOwner = this
        binding.mainViewModel = mainViewModel
        binding.mainSearchView.clearFocus()
        binding.mainSearchView.inputType = InputType.TYPE_NULL

        setEventListener()
    }

    private fun setEventListener() {
        binding.mainSearchView.setOnClickListener {
            binding.mainSearchView.clearFocus()
            val intent = Intent(this, SearchActivity::class.java)
            startActivity(intent)
        }

        binding.mainSearchView.setOnQueryTextFocusChangeListener { _, hasFocus ->
            if (hasFocus) {
                binding.mainSearchView.clearFocus()
                val intent = Intent(this, SearchActivity::class.java)
                startActivity(intent)
            }
        }

        binding.mainTextView.setOnClickListener{
            val intent = Intent(this, LckMainActivity::class.java)
            startActivity(intent)
        }
    }
}